consensus/ibft/src/integration-test/java/tech/pegasys/pantheon/consensus/ibft/support/MessageReceptionHelpers.java

```diff
@@ -52,15 +53,36 @@ public static void assertPeersReceivedExactly(
             messageMatchesExpected(rxMsgData, msg);
           });
     }
-    allPeers.forEach(p -> p.clearReceivedMessages());
+    allPeers.forEach(ValidatorPeer::clearReceivedMessages);
+  }
+
+  @SafeVarargs
+  public static void assertPeersContainsReceivedMessages(
+      final Collection<ValidatorPeer> allPeers, final SignedData<? extends Payload>... msgs) {
+    final List<SignedData<? extends Payload>> msgList = Arrays.asList(msgs);
+    allPeers.forEach(
+        node -> {
+          final List<SignedData<?>> rxMsgs =
+              node.getReceivedMessages()
+                  .stream()
+                  .map(MessageReceptionHelpers::payloadFromMessageData)
+                  .collect(Collectors.toList());
+          assertThat(rxMsgs).containsAll(msgList);
+        });
+    allPeers.forEach(ValidatorPeer::clearReceivedMessages);
```

Comment:
nit: SLIGHTLY worried this deletes all the msgs - even ones not checked....but maybe that's fine ...

Reply:
I think it's fine. I'm expecting that you will use either this message assertion or existing one.